minecraft/listener.go

```diff
@@ -172,6 +175,26 @@ func (listener *Listener) Disconnect(conn *Conn, message string) error {
 	return conn.Close()
 }
 
+// AddResourcePack adds a new resource pack to the listener's configuration.
+func (listener *Listener) AddResourcePack(pack *resource.Pack) {
+	listener.mu.Lock()
+	defer listener.mu.Unlock()
+	listener.cfg.ResourcePacks = append(listener.cfg.ResourcePacks, pack)
+}
+
+// RemoveResourcePack removes a resource pack from the listener's configuration by its UUID.
+func (listener *Listener) RemoveResourcePack(uuid string) {
+	listener.mu.Lock()
+	defer listener.mu.Unlock()
+	for i, resourcePack := range listener.cfg.ResourcePacks {
+		if resourcePack.UUID() == uuid {
+			listener.cfg.ResourcePacks[i] = nil
+			listener.cfg.ResourcePacks = append(listener.cfg.ResourcePacks[:i], listener.cfg.ResourcePacks[i+1:]...)
```

Comment:
These actions on `listener.cfg.ResourcePacks` are not (thread-)safe. When a `*Listener` is created, it should probably create a copy of the Config's slice to prevent it from keeping the same underlying array.